mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -966,11 +1047,13 @@ void mav_vehicle::send_mission_waypoint(global_pos_int wp, bool autorotate)
 
     print_verbose("New mission received\n");
 
-    // We need to toggle from AUTO to GUIDED in order to update the mission
-    // waypoints
-    set_mode(mode::GUIDED, 0);
+    // We need to toggle from AUTO to GUIDED in order to update ArduPilot
+    // mission waypoints.
+    if (this->flight_stack_type == firmware_type::APM) {
+        set_mode(mode::GUIDED, 0);
+    }
```

Comment:
set_yaw_target() and send_mission_waypoint() are not related. We need to make sure both of them change the mode to GUIDED independently.